dashboard/app/controllers/lessons_controller.rb

```diff
@@ -249,7 +249,7 @@ def cachable_request?(request)
       course = UnitGroup.get_from_cache(course_name)
       unit_position = params[:unit_position]
       raise ActiveRecord::RecordNotFound unless course && unit_position
-      unit_group_unit = UnitGroupUnit.find_by(course_id: course.id, position: unit_position)
+      unit_group_unit = UnitGroupUnit.get_with_position_from_cache(course.id, unit_position)
```

Comment:
I am not sure we need this new method either because it seems like we can get the info you need from the course cache:
```suggestion
      unit_group_unit = UnitGroup.get_from_cache(course.id).default_unit_group_units.select{|ugu| ugu.position == unit_position}
```

I could be mistaken, but I think it's worth investigating whether a solution using the existing cache works, so that we can hopefully avoid adding another caching implementation.

Reply:
Another good catch, I added unit tests for this as well. The function has been simplified to be wrapper for the code you wrote.